docs/error-handling-guidelines.md

```
@@ -0,0 +1,107 @@
+
+```go
+return &diagnostics.Err{Reason: "MissingQuota", Message: msg}
+```
+
+Currently only used for quota check failures. `Reason` must be a single CamelCase word. `Message` is free-form text written for non-expert users.
+
+## Asset Dependency Graph Error Propagation
+
+The asset store (`pkg/asset/store/store.go`) propagates errors with dependency context automatically:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)